specs/WebResourceRequested-CustomScheme.md

```diff
@@ -0,0 +1,295 @@
+            }
+            else
+            {
+                e.Response = webView.CoreWebView2.CreateWebResourceResponse(null, 404, "Not Found", L"");
+            }
+        }
+}
+
+webView.CoreWebView2.Navigate(L"https://www.example.com");
+
+// The following XHR will execute in the context of https://www.example.com page and will succeed.
+// WebResourceRequested event will be raised for this request as *.example.com is in the allowed origin list of custom-scheme.
+// Since the response header provided in WebResourceRequested handler allows all origins for CORS
+// the XHR succeeds.
+webView.CoreWebView2.ExecuteScriptAsync("var oReq = new XMLHttpRequest();\
+                                        oReq.addEventListener(\"load\", reqListener);\
+                                        oReq.open(\"GET\", \"custom-scheme://example-data.json\");\
+                                        oReq.send();");
+// The following XHR will fail because *.example.com is in the not allowed origin list of custom-scheme2.
+// The WebResourceRequested event will not be raised for this request.
+webView.CoreWebView2.ExecuteScriptAsync("var oReq = new XMLHttpRequest();\
+                                        oReq.addEventListener(\"load\", reqListener);\
+                                        oReq.open(\"GET\", \"custom-scheme-not-in-allowed-origins://example-data.json\");\
+                                        oReq.send();");
+```
+
+``` cpp
+Microsoft::WRL::ComPtr<ICoreWebView2EnvironmentOptions3> options3;
+if (options.As(&options3) == S_OK) {
+  std::vector<Microsoft::WRL::ComPtr<ICoreWebView2CustomSchemeRegistration>> schemeRegistrations;
+  
+  const WCHAR* allowedOrigins[1] = {L"https://*.example.com"};
+  schemeRegistrations.push_back(Microsoft::WRL::Make<CoreWebView2CustomSchemeRegistration>(
+          L"custom-scheme", TRUE /* isSecure*/, 1, allowedOrigins));
+  schemeRegistrations.push_back(Microsoft::WRL::Make<CoreWebView2CustomSchemeRegistration>(
+          L"custom-scheme-not-in-allowed-origins-list", TRUE /* isSecure*/, nullptr));
+  CHECK_FAILURE(options3->SetCustomSchemeRegistrations(schemeRegistrations.size(), schemeRegistrations.data());
+}
+...
+
+CHECK_FAILURE(m_webView->AddWebResourceRequestedFilter(
+        L"custom-scheme*", COREWEBVIEW2_WEB_RESOURCE_CONTEXT_ALL));
+CHECK_FAILURE(m_webView->add_WebResourceRequested(
+    Callback<ICoreWebView2WebResourceRequestedEventHandler>(
+        [this](
+            ICoreWebView2* sender,
+            ICoreWebView2WebResourceRequestedEventArgs* args) {
+        wil::com_ptr<ICoreWebView2WebResourceRequest> request;
+        wil::com_ptr<ICoreWebView2WebResourceResponse> response;
+        CHECK_FAILURE(args->get_Request(&request));
+        wil::unique_cotaskmem_string uri;
+        CHECK_FAILURE(request->get_Uri(&uri));
+        if (wcsncmp(uri.get(), L"custom-scheme", ARRAYSIZE(L"custom-scheme")-1) == 0)
```

Comment:
This sample does not match the C# sample. The C# samples accepts `custom-scheme://*`, but this one accepts `custom-scheme*://*` (and crashes on `custom-scheme:unhosted`).

Reply:
Please fix:
* C++ is correct for the filter and C# should match
* Please fix C++ to not crash.